src/mame/includes/cinemat.h

```diff
@@ -114,22 +98,66 @@ class cinemat_state : public driver_device
-	void boxingb_sound_w(uint8_t sound_val, uint8_t bits_changed);
-	void wotw_sound_w(uint8_t sound_val, uint8_t bits_changed);
-	void demon_sound_w(uint8_t sound_val, uint8_t bits_changed);
+	DECLARE_WRITE_LINE_MEMBER(spacewar_sound0_w);
+	DECLARE_WRITE_LINE_MEMBER(spacewar_sound1_w);
+	DECLARE_WRITE_LINE_MEMBER(spacewar_sound2_w);
+	DECLARE_WRITE_LINE_MEMBER(spacewar_sound3_w);
+	DECLARE_WRITE_LINE_MEMBER(spacewar_sound4_w);
+	DECLARE_WRITE_LINE_MEMBER(barrier_sound0_w);
+	DECLARE_WRITE_LINE_MEMBER(barrier_sound1_w);
+	DECLARE_WRITE_LINE_MEMBER(barrier_sound2_w);
+	DECLARE_WRITE_LINE_MEMBER(speedfrk_start_led_w);
+	DECLARE_WRITE_LINE_MEMBER(speedfrk_sound3_w);
+	DECLARE_WRITE_LINE_MEMBER(speedfrk_sound4_w);
+	DECLARE_WRITE_LINE_MEMBER(starhawk_sound0_w);
+	DECLARE_WRITE_LINE_MEMBER(starhawk_sound1_w);
+	DECLARE_WRITE_LINE_MEMBER(starhawk_sound2_w);
+	DECLARE_WRITE_LINE_MEMBER(starhawk_sound3_w);
+	DECLARE_WRITE_LINE_MEMBER(starhawk_sound4_w);
+	DECLARE_WRITE_LINE_MEMBER(starhawk_sound7_w);
+	DECLARE_WRITE_LINE_MEMBER(sundance_sound0_w);
+	DECLARE_WRITE_LINE_MEMBER(sundance_sound1_w);
+	DECLARE_WRITE_LINE_MEMBER(sundance_sound2_w);
+	DECLARE_WRITE_LINE_MEMBER(sundance_sound3_w);
+	DECLARE_WRITE_LINE_MEMBER(sundance_sound4_w);
+	DECLARE_WRITE_LINE_MEMBER(sundance_sound7_w);
+	DECLARE_WRITE_LINE_MEMBER(tailg_sound_w);
+	DECLARE_WRITE_LINE_MEMBER(warrior_sound0_w);
+	DECLARE_WRITE_LINE_MEMBER(warrior_sound1_w);
+	DECLARE_WRITE_LINE_MEMBER(warrior_sound2_w);
+	DECLARE_WRITE_LINE_MEMBER(warrior_sound3_w);
+	DECLARE_WRITE_LINE_MEMBER(warrior_sound4_w);
+	DECLARE_WRITE_LINE_MEMBER(armora_sound0_w);
+	DECLARE_WRITE_LINE_MEMBER(armora_sound1_w);
+	DECLARE_WRITE_LINE_MEMBER(armora_sound2_w);
+	DECLARE_WRITE_LINE_MEMBER(armora_sound3_w);
+	DECLARE_WRITE_LINE_MEMBER(armora_sound4_w);
+	DECLARE_WRITE_LINE_MEMBER(ripoff_sound1_w);
+	DECLARE_WRITE_LINE_MEMBER(ripoff_sound2_w);
+	DECLARE_WRITE_LINE_MEMBER(ripoff_sound3_w);
+	DECLARE_WRITE_LINE_MEMBER(ripoff_sound4_w);
+	DECLARE_WRITE_LINE_MEMBER(ripoff_sound7_w);
+	DECLARE_WRITE_LINE_MEMBER(starcas_sound0_w);
+	DECLARE_WRITE_LINE_MEMBER(starcas_sound1_w);
+	DECLARE_WRITE_LINE_MEMBER(starcas_sound2_w);
+	DECLARE_WRITE_LINE_MEMBER(starcas_sound3_w);
+	DECLARE_WRITE_LINE_MEMBER(starcas_sound4_w);
+	DECLARE_WRITE_LINE_MEMBER(solarq_sound0_w);
+	DECLARE_WRITE_LINE_MEMBER(solarq_sound1_w);
+	DECLARE_WRITE_LINE_MEMBER(solarq_sound4_w);
+	DECLARE_WRITE_LINE_MEMBER(boxingb_sound0_w);
+	DECLARE_WRITE_LINE_MEMBER(boxingb_sound1_w);
+	DECLARE_WRITE_LINE_MEMBER(boxingb_sound2_w);
+	DECLARE_WRITE_LINE_MEMBER(boxingb_sound3_w);
+	DECLARE_WRITE_LINE_MEMBER(boxingb_sound4_w);
+	DECLARE_WRITE_LINE_MEMBER(wotw_sound0_w);
+	DECLARE_WRITE_LINE_MEMBER(wotw_sound1_w);
+	DECLARE_WRITE_LINE_MEMBER(wotw_sound2_w);
+	DECLARE_WRITE_LINE_MEMBER(wotw_sound3_w);
+	DECLARE_WRITE_LINE_MEMBER(wotw_sound4_w);
+	DECLARE_WRITE_LINE_MEMBER(demon_sound4_w);
+	DECLARE_WRITE_LINE_MEMBER(qb3_sound4_w);
```

Comment:
This really looks like a case for at splitting up the all-in-one driver state into a family of derived classes, or making devices for each of the audio variants providing a consistent API.  This just looks gross.